storage/tests/unit/test_blob.py

```diff
@@ -366,7 +366,7 @@ def test__make_transport(self, fake_session_factory):
 
     def test__get_download_url_with_media_link(self):
         blob_name = 'something.txt'
-        bucket = mock.Mock(spec=[])
+        bucket = _Bucket(name='IRRELEVANT')
```

Comment:
I dissagree, pretty strongly:  having `user_project` on the bucket instances (defaulted to None) is much clearer than having to tweak each `Mock` call site to set it (and add it to the spec).